pkg/expect/expect.go

```diff
@@ -98,43 +102,76 @@ func (ep *ExpectProcess) read() {
 	printDebugLines := os.Getenv("EXPECT_DEBUG") != ""
 	r := bufio.NewReader(ep.fpty)
 	for {
-		l, err := r.ReadString('\n')
-		ep.mu.Lock()
-		if l != "" {
-			if printDebugLines {
-				fmt.Printf("%s (%s) (%d): %s", ep.cmd.Path, ep.cfg.name, ep.cmd.Process.Pid, l)
+		err := func() error {
```

Comment:
no need for anonymous function

Reply:
disagree, this makes the locking more obvious in 

```
ep.mu.Lock()
defer ep.mu.Unlock()
```

Reply:
Sounds good.